bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -176,6 +182,7 @@ type APIServer struct {
 	// certSANs sets extra Subject Alternative Names for the API Server signing cert.
 	// +optional
 	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:items:MaxLength=512
```

Comment:
I would expect the longest SAN to be a DNS name, in which case the max is 253 chars